datatypes/src/raster/raster_tile.rs

```diff
@@ -94,8 +94,25 @@ where
         time: TimeInterval,
         tile_info: TileInformation,
         data: GridOrEmpty<D, T>,
-    ) -> Self {
-        // TODO: assert, tile information xy size equals the data xy size
+    ) -> Self
+    where
+        D: GridSize,
+    {
+        debug_assert_eq!(
+            tile_info.tile_size_in_pixels.axis_size_x(),
```

Comment:
can we implement PartialEq on GridSize and compare tile_info as GridSize with data as GridSize?

Reply:
The Problem is something else. D is  just anything that is a Dimension but the query knows that it is 2D  so the types to not match. Since it is a debug assert i guess thats something for another time.